File: tests/indexing/parsing/lang/fixtures/go/Closures.go
package example

type Worker struct {
	onComplete func()
}

func Transform(nums []int) []int {
	doubler := func(x int) int { return x * 2 }

	w := Worker{}
	w.onComplete = func() {
		println("done")
	}

	register(func(v int) int {
		scaled := doubler(v)
		return scaled + 1
	})

	result := make([]int, 0, len(nums))
	for _, n := range nums {
		result = append(result, doubler(n))
	}
	return result
}

File: tests/indexing/parsing/lang/fixtures/rust/Closures.rs
struct Worker {
    on_complete: Box<dyn Fn()>,
}

pub fn run(values: &[i32]) -> Vec<i32> {
    let doubler = |x: i32| x * 2;

    let mut worker = Worker {
        on_complete: Box::new(|| {}),
    };
    worker.on_complete = || {
        println!("done");
    };

    values
        .iter()
        .map(|v| {
            let scaled = doubler(*v);
            scaled + 1
        })
        .collect()
}
